src/Microsoft.Diagnostics.EventFlow.Core/Implementations/LossReportingPropagatorBlock.cs

```diff
@@ -0,0 +1,209 @@
+                    Debug.Assert(this.target != null);
+                    return this.target.Completion;
+                }
+                else
+                {
+                    return this.completionSource.Value.Task;
+                }
+            }
+        }
+
+        public void Complete()
+        {
+            this.completed = true;
+
+            if (this.propagateCompletion)
+            {
+                Debug.Assert(this.target != null);
+                this.target.Complete();
+            }
+            else
+            {
+                this.completionSource.Value.TrySetResult(true);
+            }
+        }
+
+        public void Fault(Exception exception)
+        {
+            this.completed = true;
+
+            if (this.propagateCompletion)
+            {
+                Debug.Assert(this.target != null);
+                this.target.Fault(exception);
+            }
+            else
+            {
+                this.completionSource.Value.TrySetException(exception);
+            }
+        }
+
+        public IDisposable LinkTo(ITargetBlock<T> target, DataflowLinkOptions linkOptions)
+        {
+            Requires.NotNull(target, nameof(target));
+            if (this.target != null)
+            {
+                throw new InvalidOperationException($"{nameof(LossReportingPropagatorBlock<T>)} only supports a single target");
+            }
+
+            this.target = target;
+            if (linkOptions != null)
+            {
+                // Note DataflowLinkOptions.Append does not matter given that we are working with a single target only 
+                // and we will not enforce DataflowLinkOptions.MaxMessages
+                this.propagateCompletion = linkOptions.PropagateCompletion;
+            }
+
+            // We do not support re-linking
+            return new EmptyDisposable();
+        }
+
+        public DataflowMessageStatus OfferMessage(DataflowMessageHeader messageHeader, T messageValue, ISourceBlock<T> source, bool consumeToAccept)
+        {
+            VerifySourceInitiatedOperation(source);
+
+            if (this.completed)
+            {
+                return DataflowMessageStatus.DecliningPermanently;
+            }
+
+            var targetResponse = this.target.OfferMessage(messageHeader, messageValue, this, consumeToAccept);
+            switch(targetResponse)
+            {
+                case DataflowMessageStatus.Accepted:
+                    // Great, nothing to do
+                    break;
+
+                case DataflowMessageStatus.Declined:
+                case DataflowMessageStatus.DecliningPermanently:
+                case DataflowMessageStatus.NotAvailable:
+                    this.healthReporter.ReportThrottling();
+                    break;
+
+                case DataflowMessageStatus.Postponed:
+                    // This is a bit tricky--the message might be successfully consumed later, or it might no longer be available.
```

Comment:
Could add more comment here explain what the trick is, like BroadCastBlock will store one message, so it might successfully get the postponed message later on. But if new message arrives at the BroastCaseBlock and overrides the message, then it will fail at ConsumeMessage.